core/src/main/java/com/github/phantazmnetwork/core/gui/Gui.java

```diff
@@ -0,0 +1,286 @@
+package com.github.phantazmnetwork.core.gui;
+
+import com.github.phantazmnetwork.commons.Tickable;
+import net.kyori.adventure.text.Component;
+import net.minestom.server.entity.Player;
+import net.minestom.server.inventory.Inventory;
+import net.minestom.server.inventory.InventoryType;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.function.BiPredicate;
+
+/**
+ * Extension of {@link Inventory} designed to ease the creation of graphical user interfaces. May or may not be
+ * "dynamic". Dynamic GUIs support animations and tick all of their constituent {@link GuiItem}s.
+ */
+@SuppressWarnings("SynchronizationOnLocalVariableOrMethodParameter")
+public class Gui extends Inventory implements Tickable {
```

Comment:
couldn't AbstractInventory#setItemStack(int, ItemStack) be a problem

Reply:
could be used to add itemstacks that aren't themselves GuiItems, or change the visual of a GuiItem in a way that would get overwritten when that item is next ticked (if dynamic)

Reply:
is that sufficiently dangerous to warrant overriding

Reply:
the 2nd example sounds like bad design
I guess the first one is not too bad and could excuse not having an override

Reply:
will leave as-is for now then